x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -714,14 +777,21 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeString(reason);
             out.writeLong(unassigned);
             out.writeLong(assigned);
+            if (out.getVersion().onOrAfter(SHARD_IDS_OUTPUT_VERSION)) {
+                out.writeCollection(unassignedShardIds);
+                out.writeCollection(assignedShardIds);
+            }
         }
 
         @Override
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
             builder.startObject();
             builder.field("reason", reason);
             builder.field("unassigned", unassigned);
+            builder.field("unassigned_shard_ids", unassignedShardIds);
             builder.field("assigned", assigned);
+            builder.field("assigned_shard_ids", assignedShardIds.stream().limit(MAX_ASSIGNED_SHARD_IDS).collect(Collectors.toList()));
+            builder.field("amount_of_assigned_shards", assignedShardIds.size());
```

Comment:
Can we name this `assigned_shards` and `assigned_shards_count` instead?

```suggestion
            builder.field("assigned_shards", assignedShardIds.stream().limit(MAX_ASSIGNED_SHARD_IDS).collect(Collectors.toList()));
            builder.field("assigned_shards_count", assignedShardIds.size());
```